tests/test_resources/test_secrets/conftest.py

```diff
@@ -0,0 +1,49 @@
+import pytest
+
+import runhouse as rh
+
+from tests.conftest import init_args
+
+base_secret_values = {"secret_key": "secret_val"}
+
+provider_secret_values = {
+    "aws": {"access_key": "test_access_key", "secret_key": "test_secret_key"},
+    "azure": {"subscription_id": "test_subscription_id"},
+    "gcp": {"client_id": "test_client_id", "client_secret": "test_client_secret"},
+    "lambda": {"api_key": "test_api_key"},
+    "github": {"oauth_token": "test_oauth_token"},
+    "huggingface": {"token": "test_token"},
+    "ssh": {"public_key": "test_public_key", "private_key": "test_private_key"},
+    "sky": {"public_key": "test_public_key", "private_key": "test_private_key"},
+    "custom_provider": base_secret_values,
+}
+
+providers = provider_secret_values.keys()
+
+
+@pytest.fixture(scope="function")
+def secret(request):
+    """Parametrize over multiple secrets - useful for running the same test on multiple envs."""
+    return request.getfixturevalue(request.param)
+
+
+@pytest.fixture(scope="function")
+def custom_secret():
+    args = {"values": base_secret_values}
+    custom_secret = rh.secret(**args)
+    init_args[id(custom_secret)] = args
+    return custom_secret
+
+
+@pytest.fixture(params=providers)
```

Comment:
@rohinb2 could you take a look and see if you have ideas on how to get this working? basically I'm hoping to have a parameterized fixture across the provider secrets to avoid manually creating like 8 provider secrets and expanding them in the pytest levels, and gets a little messy with the overarching `request.getfixturevalue secret` fixture above.

currently, setting the provider params to the `provider_secret` fixture here, and then using `lazy_fixture("provider_secret")` inside the level resource mapping to retrieve fixtures for all providers. but turns out the fixture itself (not the name) is being passed into the `secret` fixture and `request.getfixturevalue` throws an error for not being able to find it.

Reply:
the error can be found here: https://github.com/run-house/runhouse/actions/runs/7106349953/job/19345567802?pr=195#step:5:382